packages/odyssey-react-mui/src/Checkbox.tsx

```diff
@@ -72,12 +65,46 @@ export type CheckboxProps = {
    * The value attribute of the Checkbox
    */
   value?: string;
-} & SeleniumProps;
+} & (
+  | {
+      /**
+       * Sets the checked state of the Checkbox
+       */
+      isChecked: boolean;
+      /**
+       * Determines whether the Checkbox is checked
+       * Should not be used if `isChecked` is used
+       */
+      isDefaultChecked?: never;
```

Comment:
We should leave this as a boolean and mark it as `@deprecated` if we intend to remove it.

This would be a breaking change and would bump us to Odyssey v2 while people are still converting their apps.